### PR TITLE
Modify the water shortage abnormal timeout valve setting and its name.

### PR DESCRIPTION
**Pull request by liubotongs**

### CODE REVIEW

File: src/devices/sonoff.ts [hunk 1]
```diff
@@ -1125,6 +1126,14 @@ const definitions: Definition[] = [
                 description: 'The water valve is in normal state, water shortage or water leakage',
                 access: 'STATE_GET',
             }),
+            binary({
+                name: 'behavior_of_water_shortage',
```
Comment (Koenkk) [inline src/devices/sonoff.ts:1130]: I think we can use a more self explaining name here, something like `auto_close`?

Comment (liubotongs) [reply src/devices/sonoff.ts:1130]: > I think we can use a more self explaining name here, something like `auto_close`?

how about 'auto_close_when_water_shortage', our product manager thinks it tells users more accurately.

Comment (Koenkk) [reply src/devices/sonoff.ts:1130]: Sounds good

**Comment by Koenkk** [comment]
Thanks!

**Comment by quensen** [comment]
How will this new feature work?

When I try to post `{ "auto_close_when_water_shortage": "ENABLE" }` to Mosquitto, I am getting this error:

`
z2m: Publish 'set' 'auto_close_when_water_shortage' to 'sonoff_7l' failed: 'Error: ZCL command 0x4711caffee4711caffee/1 customClusterEwelink.write({"lackWaterCloseValveTimeout":30}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"reservedBits":0,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
`

Same happens, when I toggle the switch button in Z2M's UI or even read the status.

I have firmware 1.0.3 on the SWV and Zigbee2MQTT version 1.40.0 commit 6fa1dab.

_Stefano_

**Comment by liubotongs** [comment]
> How will this new feature work?
> 
> When I try to post `{ "auto_close_when_water_shortage": "ENABLE" }` to Mosquitto, I am getting this error:
> 
> `z2m: Publish 'set' 'auto_close_when_water_shortage' to 'sonoff_7l' failed: 'Error: ZCL command 0x4711caffee4711caffee/1 customClusterEwelink.write({"lackWaterCloseValveTimeout":30}, {"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"reservedBits":0,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'`
> 
> Same happens, when I toggle the switch button in Z2M's UI or even read the status.
> 
> I have firmware 1.0.3 on the SWV and Zigbee2MQTT version 1.40.0 commit 6fa1dab.
> 
> _Stefano_

I'm very sorry for the trouble. Firmware will be supported in 1.0.4, currently ota firmware has applied for github merge request, please update the firmware and try again, thank you!

**Comment by quensen** [comment]
> I'm very sorry for the trouble. Firmware will be supported in 1.0.4, currently ota firmware has applied for github merge request, please update the firmware and try again, thank you!

Hi @liubotongs,
Thanks for your helpful and quick response.
I can confirm that the update to firmware 1.0.4 on the Sonoff reestablished the compatibility with the new UI feature.
I have a minor issue with the wording. Shouldn't it be "ENABLED" and "DISABLED" (instead of "ENABLE" and "DISABLE") as it is also a status. In fact, I would recommend "ON" and "OFF", so it doesn't need a conversion.

Best regards,
Stefan

**Comment by N3rdix** [comment]
Ran into the same issue. I therefore created a [documentation PR ](https://github.com/Koenkk/zigbee2mqtt.io/pull/3027)to give a hint...

**Comment by JayJay-Master** [comment]
Where can I get the firmware version 1.0.4. My Zigbee2MQTT installation shows only 1.0.3 als the latest version on OTA.

**Comment by N3rdix** [comment]
> Where can I get the firmware version 1.0.4. My Zigbee2MQTT installation shows only 1.0.3 als the latest version on OTA.

Mine showed up in Z2M directly after pairing. Did you try pressing the button `check for updates` in the OTA settings and simultaneously pressing the button the device to wake it?

**Comment by JayJay-Master** [comment]
Yes! Pressed button on the device to wake it up and then Check for Update. Answer: info 2024-10-03 13:40:36 No update available for 'Wasserventil WM'

**Comment by N3rdix** [comment]
Strange, as the file is there (https://github.com/Koenkk/zigbee-OTA/pull/549).
Basic question, but not sure if it is a prerequisite for the OTA Updates: you have the recent Version of Z2M installed (I had 1.39 at Installation)

**Comment by JayJay-Master** [comment]
I now have updated my Zigbee2MQTT docker container to version 1.40.2, but OTA still shows no firmware update for Sonoff SWV. with actual 1.0.3 it says "No update available for ...". Strange!
Is there any possibility to pull the update manually?

**Comment by N3rdix** [comment]
You can have a look at the [documentation ](https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#update-to-latest-firmware)and see if any of these options work. Although it is really weird why you don't see the update at all...

**Comment by JayJay-Master** [comment]
I already  found the command in the documentation, but how to find software_build_ids and date_codes?
The OTA dashboard alctually shows firmware 1.0.3 from 20240705.

**Comment by N3rdix** [comment]
you refer to MQTT topic, right? that's just the resonse you get, sending only needs the deviceID.
But I guess this won't do anything different than updating via the Z2M frontend... As a last resort I would go for the local OTA index and firmware topic I linked

**Comment by JayJay-Master** [comment]
When I send the command for checking updates with MQTT Explorer the responce is only 
`{
  "id": "Wasserventil WM"
}`

I would like to know, if I could start the update for latest firmware version manually with publishing to topic zigbee2mqtt/bridge/request/device/ota_update/update and the following payload:
`{"data":{"id": "Wasserventil WM","from":{"software_build_id":1,"date_code":"20240705"},"to":{"software_build_id":2,"date_code":"20190102"}},"status":"ok"}`
but therefore I'm missing the two software_build_ids and the date_codes for version 1.0.3 and 1.0.4. Could you please provide me with this missing information, to test if it would work.

**Comment by JayJay-Master** [comment]
Or is the software_build_id simply the firmware version, i.e. 1.0.3 or 1.0.4 and date code the file date?

**Comment by JayJay-Master** [comment]
Maybe I found a reason in the log file, why the device ist not updating:

`[2024-10-08 15:59:54] info: 	z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/Wasserventil WM', payload '{"auto_close_when_water_shortage":"DISABLE","battery":100,"current_device_status":"water_shortage","cyclic_quantitative_irrigation":{"irrigation_capacity":0,"irrigation_interval":0},"cyclic_timed_irrigation":{"irrigation_duration":0,"irrigation_interval":0},"device":{"applicationVersion":16,"dateCode":"20240705","friendlyName":"Wasserventil WM","hardwareVersion":16,"ieeeAddr":"0x0ceff6fffe17e385","manufacturerID":4742,"manufacturerName":"SONOFF","model":"SWV","networkAddress":39522,"powerSource":"Battery","softwareBuildID":"1.0.3","type":"EndDevice","zclVersion":8},"last_seen":"2024-10-08T13:59:54.566Z","linkquality":255,"state":"OFF","update":{"installed_version":4100,"latest_version":4100,"state":"idle"}}'
`

From these log entry you can see, that installed version is 4100. From [The sonoff smart water valve equipment has added automatic valve clos…](https://github.com/Koenkk/zigbee-OTA/pull/549/commits/170a3d3f499d0d9d40c905bc6bb717872a7d6d70#top) you can see, that 4100 corresponds to firmware version 1.0.4 whereas firmware version 1.0.3 corresponds to 4099. 
How can I check, which firmware is already installed on my device.

**Pull request merged**